app/routers/dataresource.py

```diff
@@ -37,91 +53,157 @@ async def create_dataresource(
+        # Resource strategy
+        session_update = create_strategy("resource", config).get(session=session_data)
+        if session_update and session_id:
+            await _update_session(session_id, session_update, cache)
+
+    elif config.downloadUrl and config.mediaType:
+        # Download strategy
+        session_update = create_strategy("download", config).get(session=session_data)
+        if session_update and session_id:
+            session_data = await _update_session(session_id, session_update, cache)
+
+        # Parse strategy
+        session_update = create_strategy("parse", config).get(session=session_data)
+        if session_update and session_id:
+            await _update_session(session_id, session_update, cache)
+
     else:
-        raise HTTPException(
-            status_code=404,
-            detail="Missing downloadUrl/mediaType or accessUrl/accessService identifier",
-        )
-    return output
+        raise httpexception_422_resource_id_is_unprocessable(resource_id)
+
+    return GetResourceResponse(**session_update)
 
 
-@router.post("/{resource_id}/initialize")
+@ROUTER.post(
+    "/{resource_id}/initialize",
+    response_model=InitializeResourceResponse,
+    responses={
+        status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {"model": HTTPValidationError},
+    },
+)
 async def initialize_dataresource(
     resource_id: str,
     session_id: Optional[str] = None,
     cache: Redis = Depends(depends_redis),
-) -> dict:
-    """Initialize data resource"""
-    resource_info_json = json.loads(await cache.get(resource_id))
-    resource_config = ResourceConfig(**resource_info_json)
-
-    if resource_config.accessUrl and resource_config.accessService:
-        strategy = create_strategy("resource", resource_config)
-    elif resource_config.downloadUrl and resource_config.mediaType:
-        strategy = create_strategy("download", resource_config)
-    else:
-        raise HTTPException(
-            status_code=404,
-            detail="Missing downloadUrl/mediaType or accessUrl/accessService identifier",
+) -> InitializeResourceResponse:
+    """Initialize data resource."""
+    if not await cache.exists(resource_id):
+        raise httpexception_404_item_id_does_not_exist(resource_id, "resource_id")
+    if session_id and not await cache.exists(session_id):
+        raise httpexception_404_item_id_does_not_exist(session_id, "session_id")
+
+    # Using `.construct` here to avoid validation of already validated data.
+    # This is a slight speed up.
+    # https://pydantic-docs.helpmanual.io/usage/models/#creating-models-without-validation
+    config = ResourceConfig.construct(**json.loads(await cache.get(resource_id)))
+
+    session_data: "Optional[Dict[str, Any]]" = (
+        None if not session_id else json.loads(await cache.get(session_id))
+    )
+
+    if config.accessUrl and config.accessService:
```

Comment:
This is complicated, because it is possible to define all properties for a resource (accessUrl, accessService, downloadUrl AND mediaType), in which case it makes sense to use the download strategy. I would suggest to check for downloadUrl/mediaType first for download, and then all other permutations after.

Reply:
Okay - so switch the priority essentially, favoring download strategies?

Reply:
I'll create an issue for this and then it can be tackled separately.

Reply:
Issue #56 has been created for this.